packages/babel-helper-remap-async-to-generator/src/index.js

```diff
@@ -145,6 +145,15 @@ function plainFunction(path: NodePath, callId: Object) {
     ]);
     declar._blockHoist = true;
 
+    if (path.parentPath.isExportDefaultDeclaration()) {
+      // change the path type so that replaceWith() does not wrap
+      // the identifier into an expressionStatement
+      path.type = "FunctionExpression";
+      path.parentPath.insertBefore(declar);
+      path.replaceWith(t.identifier(asyncFnId.name));
```

Comment:
It still won't be exported up front, as `asyncId` will be in TDZ until the module body finishes running, but that problem is completely unavoidable.

It does need to be exported with `{asyncId as default}` anyway to keep live binding.